mkosi/__init__.py

```diff
@@ -1873,111 +1873,96 @@ def make_rpm_list(args: MkosiArgs, packages: Set[str], do_run_build_script: bool
     return packages
 
 
-def clean_dnf_metadata(root: Path, always: bool) -> None:
-    """Remove dnf metadata if /bin/dnf is not present in the image
+def flatten(lists: Iterable[Iterable[T]]) -> Sequence[T]:
```

Comment:
We tend to not widen the type when used as a return type, so this should return `List[T]` instead of `Sequence[T]`